relays/bin-substrate/src/chains/kusama.rs

```diff
@@ -32,13 +37,15 @@ use crate::cli::{
 pub(crate) const SYSTEM_REMARK_CALL_WEIGHT: Weight = 2 * 1_345_000;
 
 impl CliEncodeCall for Kusama {
-	fn encode_call(call: &Call) -> anyhow::Result<Self::Call> {
+	fn encode_call(call: &Call) -> anyhow::Result<EncodedOrDecodedCall<Self::Call>> {
 		Ok(match call {
+			Call::Raw { data } => EncodedOrDecodedCall::Encoded(data.0.clone()),
```

Comment:
so this fn only returns `EncodedOrDecodedCall::Encoded` variant for `Call::Raw`, right?

Reply:
That's correct. Why is it important for us - we can't maintain the whole copy of Polkadot/Kusama `enum Call` here, in substrate code. Instead, we allow to encode call externally and use this `raw` variant. We can't decode it here, because for most of calls, decode will fail (except for this `remark`). So yes - it only returns `EncodedOrDecodedCall::Encoded` for `Call::Raw`